R/ci.R

```diff
@@ -1,16 +1,74 @@
 #' Continuous integration setup and badges
 #'
 #' Sets up continuous integration (CI) services for an R package that is
-#' developed on GitHub. CI services can run `R CMD check` automatically on
-#' various platforms, triggered by each push or pull request. These functions
-#' * Add service-specific configuration files and add them to `.Rbuildignore`.
-#' * Activate a service or give the user a detailed prompt.
-#' * Provide the markdown to insert a badge into README.
-#'
+#' developed on GitHub with CI-agnostic support by \pkg{tic}.
+#' CI services can run `R CMD check` automatically on
+#' various platforms, triggered by each push or pull request. This function
+#' * Adds service-specific configuration files and adds them to `.Rbuildignore`.
+#' * Activates a service or gives the user a detailed prompt.\cr
 #' @name ci
-#' @aliases NULL
 NULL
 
+#' @section `use_ci()`:
+#' By default the CI-services "Travis" (Linux) and "Appveyor"
+#' (Windows) will be set up. Basic `.travis.yml` and `appveyor.yml` files are
+#' added to the top-level directory of a package.
+#'
+#' This function is aimed at supporting the most common use cases.
+#' Users who require more control are advised to manually call the individual
+#' functions.
+#' @importFrom travis travis_set_pat
+#' @param path `[string]`\cr
+#'   The path to the repo to prepare.
+#' @param quiet `[flag]`\cr
+#'   Less verbose output? Default: `FALSE`.
+#' @param services `[character]`\cr
+#'   CI services to add.
+#' @export
+#' @rdname ci
+use_ci <- function(path = ".", quiet = FALSE,
+                   services = c("travis", "appveyor")) {
+  #' @details
+  #' The following steps will be run (`use_ci()` only):
+  withr::with_dir(path, {
+    #' 1. If necessary, create a GitHub repository via [use_github()]
+    use_github()
+
+    if ("travis" %in% services) {
+      #' 1. Enable Travis via [travis_enable()]
+      travis_enable()
+
+      #' 1. Create a default `.travis.yml` file
+      #'    (overwrite after confirmation in interactive mode only)
+      use_template(
+        "travis.yml",
+        ".travis.yml",
+        ignore = TRUE
+      )
+    }
+
+    repo_type <- detect_repo_type()
+
+    if ("appveyor" %in% services) {
+      #' 1. Create a default `appveyor.yml` file
+      #'    (depending on repo type, overwrite after confirmation
+      #'    in interactive mode only)
+      if (travis:::needs_appveyor(repo_type)) travis:::use_appveyor_yml() #FIXME: Export function in travis
```

Comment:
`needs_appveyor()` can be just copied here.

Do we have an `appveyor.yml` template here?

Reply:
Yes, I modified the existing one.